tests/Unit/Repositories/BotRepositoryTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Tests\Unit\Repositories;
+
+use Auth;
+use Mockery;
+use Tests\TestCase;
+use App\Models\Bot;
+use App\Repositories\Eloquents\BotRepository;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class BotRepositoryTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * get model
+     *
+     * @return void
+     */
+    public function testGetModel()
+    {
+        $botRepository = new BotRepository;
+
+        $data = $botRepository->getModel();
+        $this->assertEquals(Bot::class, $data);
+    }
+
+    /**
+     * get all list bot by user
+     *
+     * @return void
+     */
+    public function testGetAllByUser()
+    {
+        $botRepository = new BotRepository;
+        $botLists = factory(Bot::class, 10)->create(['name' => 'test get all bot by user']);
+        $user = $botLists[0]->user;
+        Auth::shouldReceive('user')->once()->andReturn($user);
+
+        $botRepository->getAllByUser();
+        $this->assertDatabaseHas('bots', ['name' => 'test get all bot by user']);
```

Comment:
test_case này mình phải test data trả về của `$botRepository->getAllByUser()` có đúng với như mình mong muốn ko thì đúng hơn chứ nhỉ?
còn ở đây e test trong db có gì rồi

Reply:
$botLists = factory(Bot::class, 10)->create(['name' => 'test get all bot by user']); 
chỗ này em tạo dữ liệu  insert vào
$botRepository->getAllByUser();
        $this->assertDatabaseHas('bots', ['name' => 'test get all bot by user']);
chỗ này em test hàm getall xem lấy ra xem có dữ liệu đó k